search_gov_crawler/search_gov_spiders/pipelines.py

```diff
@@ -4,47 +4,87 @@
+            self.file_path = self.base_file_name
+            self.current_file = open(self.file_path, "w", encoding="utf-8")
+        else:
+            self.urls_batch = []
 
     def process_item(self, item, _spider):
-        """Checks that the file is not at max size.
-        Adds it to the file if less, or creates a new file if too large."""
+        """Process item either by writing to file or by posting to API."""
+
+        line = item.get("url", "") + "\n"
+        line_size = len(line.encode('utf-8'))
 
-        line = item["url"]
-        self.current_file_size += 1
-        next_file_size = self.current_file_size + len(line)
-        if self.paginate and next_file_size > self.max_file_size:
-            self.short_file.close()
-            new_name = str(self.parent_file_path / f"output/all-links{self.file_number}.csv")
-            os.rename(self.base_path_name, new_name)
-            self.file_number = self.file_number + 1
-            self.short_file = open(self.base_path_name, "w", encoding="utf-8")
-            self.current_file_size = 0
-
-        self.short_file.write(line)
-        self.short_file.write("\n")
-        self.current_file_size = self.current_file_size + len(line)
+        # If API URL is set, batch URLs and send a POST request when max size is reached
+        if self.api_url:
+            self.urls_batch.append(item.get("url", ""))
+            if self._is_batch_too_large(line_size):
+                self._post_urls()
+        # Otherwise, write to file and rotate if needed
+        else:
+            self.current_file.write(line)
+            if self._is_file_too_large(line_size):
+                self._rotate_file()
 
         return item
 
+    def _is_batch_too_large(self, new_entry_size):
+        current_batch_size = sum(len(url.encode('utf-8')) for url in self.urls_batch)
+        return (current_batch_size + new_entry_size) > self.MAX_FILE_SIZE_BYTES
+
+    def _is_file_too_large(self, new_entry_size):
+        self.current_file.flush()
+        current_file_size = self.file_path.stat().st_size
+        return (current_file_size + new_entry_size) > self.MAX_FILE_SIZE_BYTES
+
+    def _rotate_file(self):
+        """Close current file, rename it, and open a new one for continued writing."""
+        self.current_file.close()
+        new_file_path = self.parent_file_path / f"output/all-links-{self.file_number}.csv"
+        os.rename(self.file_path, new_file_path)
+        self.file_number += 1
+        self.current_file = open(self.file_path, "w", encoding="utf-8")
+
+    def _post_urls(self):
+        """Send a POST request with the batch of URLs if any exist."""
+        if self.urls_batch:
+            try:
+                response = requests.post(self.api_url, json={"urls": self.urls_batch})
+                response.raise_for_status()
+                print(f"Successfully posted {len(self.urls_batch)} URLs to {self.api_url}.")
```

Comment:
I think you can probably access a logger here... can this be a `logger.info()` instead of a `print`?